plugins/node/opentelemetry-instrumentation-connect/src/instrumentation.ts

```diff
@@ -180,4 +212,33 @@ export class ConnectInstrumentation extends InstrumentationBase<Server> {
       return original.apply(this, args as UseArgs2);
     };
   }
+
+  public _patchHandle(original: Server['handle']): Server['handle'] {
+    const instrumentation = this;
+    return function (this: Server): ReturnType<Server['handle']> {
+      const [reqIdx, outIdx] = [0, 2];
+      const req = arguments[reqIdx] as PatchedRequest;
+      const out = arguments[outIdx];
+      addNewStackLayer(req);
+
+      function completeStack() {
+        req[_LAYERS_STORE_PROPERTY].pop();
+      }
```

Comment:
since above we have 2 function to manipulate the `req[_LAYERS_STORE_PROPERTY]` property, it might be readable to also add this function there, something like:
```js
const addNewStackLayer = (request: PatchedRequest) => {
...
}

const replaceCurrentStackRoute = (request: PatchedRequest, value?: string) => {
...
}

const popLastRoute = (request: PatchedRequest) => {
...
}

```

Reply:
instead of making popLastRoute a dedicated method, I make it a return function from addNewStackLayer.

That way we can ensure we can only pop the correct stack's item